packages/ensnode-sdk/src/ensindexer/indexing-status/types.ts

```diff
@@ -1,514 +1,741 @@
 export const IndexingStrategyIds = {
+  /**
+   * Represents that the indexing of events across all indexed chains will
+   * proceed in a deterministic "omnichain" ordering by block timestamp, chain ID,
+   * and block number.
+   *
+   * This strategy is "deterministic" in that the order of processing multichain indexed
+   * events and each resulting indexed data state transition recorded in ENSDb is always
+   * the same for each ENSIndexer instance operating with an equivalent
+   * `ENSIndexerConfig` and ENSIndexer version. However it also has the drawbacks of:
+   * - increased indexing latency that must wait for the slowest indexed chain to
+   *   add new blocks or to discover new blocks through the configured RPCs.
+   * - if any indexed chain gets "stuck" due to chain or RPC failures, all indexed chains
+   *   will be affected.
+   */
   Omnichain: "omnichain",
 } as const;
 
+/**
+ * The derived string union of possible {@link IndexingStrategyIds}.
+ */
+// TODO: rename to `MultichainIndexingStrategyId` (singular!)
 export type IndexingStrategyIds = (typeof IndexingStrategyIds)[keyof typeof IndexingStrategyIds];
 
 /**
- * Current Indexing Projection
- *
- * NOTE:
- * - This is "Current..." because it is the only place in
- *   the indexing status data model that should have a relationship with "now".
- * - The above is important as we want to be able to cache
- *   `OmnichainIndexingSnapshot` and then generate
- *   - `CurrentIndexingProjectionOmnichain` as a function of:
- *      - An omnichain indexing snapshot (from cache)
- *      - "now" (aka "realtime")
- * - This also includes an explicit field for `strategy` which provides
- *   more "future-proofing" and explicitly identifies how this data model is
- *   specific to omnichain indexing -- the data model and invariants would be
- *   different for other indexing strategies.
+ * Multichain indexing status snapshot when the `strategy` is
+ * {@link MultichainIndexingStrategyId.Omnichain}.
  *
  * Invariants:
- * - `realtime` is always >= `snapshot.snapshotTime`
+ * - `strategy` is always {@link MultichainIndexingStrategyId.Omnichain}.
+ * - `slowestChainIndexingCursor` is always equal to
+ *   `omnichainSnapshot.omnichainIndexingCursor`.
+ * - `snapshotTime` is always >= the "highest known block timestamp", defined as the max of:
+ *     - the `slowestChainIndexingCursor`.
+ *     - the `config.startBlock.timestamp` for all indexed chains.
+ *     - the `config.endBlock.timestamp` for all indexed chains with a `config.configType` of
+ *       {@link ChainIndexingConfigTypeIds.Definite}.
+ *     - the `backfillEndBlock.timestamp` for all chains with `chainStatus` of
+ *       {@link ChainIndexingStatusIds.Backfill}.
+ *     - the `latestKnownBlock.timestamp` for all chains with `chainStatus` of
+ *       {@link ChainIndexingStatusIds.Following}.
  */
-export type CurrentIndexingProjectionOmnichain = {
+export interface MultichainIndexingStatusSnapshotOmnichain {
```

Comment:
I'm not sure if we really need to wrap `OmnichainIndexingSnapshot` with `MultichainIndexingStatusSnapshotOmnichain`. Why not compose the _strategy_ and _slowestChainIndexingCursor_ field into to `OmnichainIndexingSnapshot` type?

In other words, I'd recommend not go build deep hierarchies (wrapping a type with another type), but rather focus on composition (having a type to implement relevant interfaces).